src/jdk.crypto.cryptoki/share/classes/sun/security/pkcs11/P11Key.java

```diff
@@ -115,17 +115,15 @@
     }
 
     P11Key(String type, Session session, long keyID, String algorithm,
-            int keyLength, CK_ATTRIBUTE[] attributes) {
+            int keyLength, CK_ATTRIBUTE[] attrs) {
         this.type = type;
         this.token = session.token;
         this.algorithm = algorithm;
         this.keyLength = keyLength;
         boolean tokenObject = false;
         boolean sensitive = false;
         boolean extractable = true;
-        int n = (attributes == null) ? 0 : attributes.length;
-        for (int i = 0; i < n; i++) {
-            CK_ATTRIBUTE attr = attributes[i];
+        for (CK_ATTRIBUTE attr : attrs) {
```

Comment:
Just noticed this: add a check for non-null attrs? If null, skip the for-loop.

Reply:
I've noticed that when reviewing your Webrev.01 but thought that the change was on purpose because of the P11Key constructor callers, which do not currently pass null. Anyways, I'll add the null check again just in case.